### PR TITLE
Fixes #5050

### LINKED ISSUE #5050: Undefined is not an object (evaluating 'rows[j][k].toString')
When I'm debugging in chrome and I click on a menu item that I've made, it runs the function like I coded. When I'm NOT debugging, I get 'Undefined is not an object (evaluating 'rows[j][k].toString')'.

https://youtu.be/6nsPr-2T6QE

https://github.com/natdm/PanResponderApp/blob/master/FullSwipe.ios.js#L172-L204

### PR DESCRIPTION
**Pull request by mihirsoni**
Fixes the issue #5050.

### CODE REVIEW

File: packager/react-packager/src/Resolver/polyfills/console.js [hunk 1]
```diff
@@ -423,13 +423,14 @@
       columns.forEach(function(k, i) {
         columnWidths[i] = k.length;
         for (var j = 0; j < rows.length; j++) {
-          var cellStr = rows[j][k].toString();
-          stringRows[j] = stringRows[j] || [];
-          stringRows[j][i] = cellStr;
-          columnWidths[i] = Math.max(columnWidths[i], cellStr.length);
+          if(rows[j][k] != undefined && rows[j][k] != null){
```
Comment (ide) [inline packager/react-packager/src/Resolver/polyfills/console.js]: Change this to just != null

Comment (mihirsoni) [inline packager/react-packager/src/Resolver/polyfills/console.js]: @ide  I have updated the PR.

Thanks for the review.

**Comment by brentvatne** [comment]
@mihirsoni - can you provide a test plan? basically an example of something that didn't work before, and works now with the change.

**Comment by ide** [comment]
To add - I think this still crashes on some edge cases... may want to add comprehensive unit tests.

**Comment by mihirsoni** [comment]
@brentvatne  Sure.

Please look at ```var rows ```
Description is provided in comment 

```js
/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 */
'use strict';

var React = require('react-native');
var {
  AppRegistry,
  StyleSheet,
  Text,
  View,
} = React;
/*In this case , without this pull request 
it will break and throw an undefined  exception.
With this pull request it will check and it console all the Colors.
*/
var rows = [{color: 'red'}, {color:'green',height: '10'} ];
var testConsoleTable = React.createClass({
  render: function() {			  
	console.table(rows);
    return (
      <View style={styles.container}>
        <Text style={styles.welcome}>
          Welcome to React Native!
        </Text>
        <Text style={styles.instructions}>
          To get started, edit index.ios.js
        </Text>
        <Text style={styles.instructions}>
          Press Cmd+R to reload,{'\n'}
          Cmd+D or shake for dev menu
        </Text>
      </View>
    );
  }
});

var styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
  instructions: {
    textAlign: 'center',
    color: '#333333',
    marginBottom: 5,
  },
});

AppRegistry.registerComponent('testConsoleTable', () => testConsoleTable);

```

**Comment by mihirsoni** [comment]
@ide  Yup correct. I have [commented] (https://github.com/facebook/react-native/issues/5050#issuecomment-168299840) on an issue , please let me know if you would like me to implement that.

currently it gets keys from the [first row] (https://github.com/facebook/react-native/blob/4cb04315e7de835b52cbdc2ae3072ddaadb9283b/packager/react-packager/src/Resolver/polyfills/console.js#L417)  only.

in the following example , in second row it won't be able to find color so it will break while joining the row.
```javascript 
var rows = [{color: 'red'}, {height: '10'} ];
```

**Comment by satya164** [comment]
@mihirsoni Adding a unit test will be cool :)

**Comment by ide** [comment]
Can you implement a bug-free solution?

**Comment by mihirsoni** [comment]
@satya164  I tried to add test , but couldn't find proper guide to do that.
Could you please help ? 
Not able to run ```xCodeRunTest.sh```

**Comment by satya164** [comment]
@mihirsoni Isn't this just JavaScript? You can write a test and run it with jest.

**Comment by mihirsoni** [comment]
@ide I tested it and seems it covers most of the edge cases. Also formatted the output which was little buggy.
Thanks for your inputs.

**Comment by mihirsoni** [comment]
Something went wrong with this pull request closing it.
